backend/src/routes/admin/controller.ts

```diff
@@ -232,3 +232,148 @@ export const getAdminStatsController = async (req: Request, res: Response) => {
     });
   }
 };
+
+export const getPostByIdController = async (req: Request, res: Response) => {
+  try {
+    const postId = req.params.id;
+
+    const post = await prisma.post.findFirst({
+      where: {
+        id: postId,
+      },
+      select: {
+        id: true,
+        title: true,
+        codeSnippet: true,
+        jsCodeSnippet: true,
+        description: true,
+        tags: true,
+      },
+    });
+
+    if (!post) {
+      return res.status(404).json({
+        message: "No such post exists!",
+      });
+    }
+
+    res.status(200).json({
+      post
+    });
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({
+      error: "Internal server error!",
+    });
+  }
+};
+
+export const updatePostController = async (req: UserAuthRequest, res: Response) => {
+  try {
+    const userId = req.userId;
+    const { postId } = req.params;
+    const { title, description, codeSnippet, jsCodeSnippet, tags } = req.body;
+
+    if (!userId) {
+      return res.status(403).json({ error: "Invalid admin" });
+    }
+
+    const user = await prisma.user.findFirst({
+      where: {
+        id: userId,
+      },
+      select: {
+        isAdmin: true,
+      },
+    });
+
+    if (!user?.isAdmin) {
+      return res.status(403).json({
+        error: "User is not admin!",
+      });
+    }
+
```

Comment:
This user admin check is not needed since it is already done in isAdmin middleware

Reply:
@VaibhavArora314 I was also not doing this but I taught lets do like the user post contoller approach, My bad but i removed that check now 👍